src/overlays/actors/ovl_En_Test7/z_en_test7.c

```diff
@@ -47,209 +48,193 @@ ActorInit En_Test7_InitVars = {
+    f32 velocityXZ;
 
-    if (arg2 != 0) {
-        Math_CosS(sp26);
-        arg0->unk_08.x = arg0->unk_08.x;
-        arg0->unk_08.y += (Rand_ZeroOne() * 100.0f) - 20.0f;
-        Math_SinS(sp26);
-        arg0->unk_08.z = arg0->unk_08.z;
+    feather->pos = *pos;
+
+    if (isPosYRand) {
+        feather->pos.x += Math_CosS(yaw) * 0.0f;
+        feather->pos.y += (Rand_ZeroOne() * 100.0f) - 20.0f;
+        feather->pos.z += Math_SinS(yaw) * 0.0f;
     }
 
-    sp20 = (Rand_ZeroOne() * 4.0f) + 2.0f;
+    velocityXZ = (Rand_ZeroOne() * 4.0f) + 2.0f;
+
+    feather->velocity.x = Math_CosS(yaw) * velocityXZ;
+    feather->velocity.y = Rand_ZeroOne();
+    feather->velocity.z = Math_SinS(yaw) * velocityXZ;
 
-    arg0->unk_14 = Math_CosS(sp26) * sp20;
-    arg0->unk_18 = Rand_ZeroOne();
-    arg0->unk_1C = Math_SinS(sp26) * sp20;
+    feather->accel.x = 0.0f;
+    feather->accel.y = 0.0f;
+    feather->accel.z = 0.0f;
 
-    arg0->unk_20 = 0.0f;
-    arg0->unk_24 = 0.0f;
-    arg0->unk_28 = 0.0f;
-    arg0->unk_2C = 0.25f;
-    arg0->unk_30.x = Rand_ZeroOne() * 0x10000;
-    arg0->unk_30.y = Rand_ZeroOne() * 0x10000;
-    arg0->unk_30.z = Rand_ZeroOne() * 0x10000;
-    arg0->unk_04 = 60;
+    feather->scale = 0.25f;
+    feather->rot.x = Rand_ZeroOne() * 0x10000;
+    feather->rot.y = Rand_ZeroOne() * 0x10000;
+    feather->rot.z = Rand_ZeroOne() * 0x10000;
+    feather->unk_04 = 60;
 
     if (Rand_ZeroOne() < 0.9f) {
-        arg0->unk_00 = 1;
-        arg0->unk_30.x = Rand_ZeroOne() * 0x10000;
-        arg0->unk_30.y = Rand_ZeroOne() * 0x10000;
-        arg0->unk_30.z = Rand_ZeroOne() * 0x10000;
-        arg0->unk_36 = 0;
-        arg0->unk_38 = 0;
-        arg0->unk_3A = 0;
+        feather->type = OWL_WARP_FEATHER_TYPE_1;
+        feather->rot.x = Rand_ZeroOne() * 0x10000;
+        feather->rot.y = Rand_ZeroOne() * 0x10000;
+        feather->rot.z = Rand_ZeroOne() * 0x10000;
+        feather->angularVelocity.x = 0;
+        feather->angularVelocity.y = 0;
+        feather->angularVelocity.z = 0;
     } else {
-        arg0->unk_00 = 2;
-        arg0->unk_30.x = 0;
-        arg0->unk_30.y = 0;
-        arg0->unk_30.z = Rand_ZeroOne() * 5000.0f;
-        arg0->unk_36 = 0;
-        arg0->unk_38 = (Rand_ZeroOne() * 8000.0f) + 2000.0f;
+        feather->type = OWL_WARP_FEATHER_TYPE_2;
+        feather->rot.x = 0;
+        feather->rot.y = 0;
+        feather->rot.z = Rand_ZeroOne() * 5000.0f;
+        feather->angularVelocity.x = 0;
+        feather->angularVelocity.y = (Rand_ZeroOne() * 8000.0f) + 2000.0f;
         if (Rand_ZeroOne() > 0.5f) {
-            arg0->unk_38 = -arg0->unk_38;
+            feather->angularVelocity.y = -feather->angularVelocity.y;
         }
-        arg0->unk_3A = 0;
+        feather->angularVelocity.z = 0;
     }
 }
 
-void func_80AF0C30(EnTest7Struct2* arg0, Vec3f* arg1, s32 arg2) {
-    static s32 D_80AF3410 = 0;
+void EnTest7_RequestFeather(OwlWarpFeather* feathers, Vec3f* pos, s32 isPosYRand) {
```

Comment:
Maybe this is just me, but when I see the word "request," my first thought is that the function can potentially fail to whatever it's requesting. In this case, though, this function can't fail to add a feather; if it doesn't add it in the loop, it will literally overwrite an existing feather.

I'm trying to think of a replacement for this name, though, and I'm struggling because `AddFeather` is already taken. Maybe the existing `AddFeather` can be `InitFeather`, and then this can be `AddFeather`? Though then `InitFeather` is almost identical to `InitFeathers`, so it's still not great. What do you think?

Reply:
That's true, this is a tricky one. When it comes to feathers or effects in general, what's you're opinion on `AddEffect` vs `InitEffect`? Once I'm done the repo sweep of anim cleanups, I want to do something similar for effects and make things consistent like `Add/Init` for example. 

This function is more like `EnTest7_AddAndChooseFeather` since this function primarily chooses which feather index to use, then calls `EnTest7_AddFeather` which is your more typical `AddEffect`/`InitEffect`

Reply:
I think I like `InitEffect`, because we tend to use `UpdateEffects` for the function that updates them, and keeping the symmetry with actors is nice to me. As for your second point, `AddAndChooseFeather` is fine with me for this one.